### PR TITLE
👪 Groups modifier should log error events for dynamic groups

### PR DESCRIPTION
**Pull request by rowanseymour**
* Move the check for dynamic groups in add/remove actions to the modifier (#865)
 * Make modifier tests snapshot based

### CODE REVIEW

File: flows/actions/base.go [hunk 1]
```diff
@@ -297,11 +297,7 @@ func resolveGroups(run flows.FlowRun, references []*assets.GroupReference, stati
 		}
 
 		if group != nil {
-			if staticOnly && group.IsDynamic() {
```
Comment (rowanseymour) [inline flows/actions/base.go]: nicely there's no duplication required.. we can just move this down to the modifier

File: flows/actions/modifiers/testdata/channel.json [hunk 2]
```diff
@@ -58,11 +58,11 @@
         "contact_after": {
             "uuid": "5d76d86b-3bb9-4d5a-b822-c9d86f5d8e4f",
             "name": "Bob",
+            "created_on": "2018-06-20T11:40:30.123456789Z",
             "urns": [
-                "tel:+12065551212?channel=3a05eaf5-cb1b-4246-bef1-f277419c83a7&id=123",
+                "tel:+12065551212?channel=3a05eaf5-cb1b-4246-bef1-f277419c83a7\u0026id=123",
```
Comment (nicpottier) [inline flows/actions/modifiers/testdata/channel.json]: I was seeing this too.. for our generic tests should we go the extra mile and do: https://github.com/golang/go/commit/ab52ad894f453a02153fb2bc106d08c47ba643b6

Comment (rowanseymour) [reply flows/actions/modifiers/testdata/channel.json]: I'm confused.. all of this JSON is marshaled by goflow's own encoder which disables at https://github.com/nyaruka/goflow/blob/master/utils/jsonx/json.go#L38 .. in theory

Comment (nicpottier) [reply flows/actions/modifiers/testdata/channel.json]: Oh hah (and hrmm)

Comment (rowanseymour) [reply flows/actions/modifiers/testdata/channel.json]: Fixed now - had to call the custom `jsonx.Marshal` all down the chain of marshals involved in marshalling a contact

**Pull request merged**